### PR TITLE
enhancing/fixing #389

### PR DESCRIPTION
**Pull request by MaximilianMeister**
#389 didn't work as expected. I opened this PR to trigger a gating, as @vuntz is likely not available

this hopefully fixes the upgrade CI and would be the better solution for https://github.com/SUSE-Cloud/automation/pull/944

### CODE REVIEW

**Comment by aplanas** [comment]
So we discarded to increase the priority in alternatives inside the package?

**Comment by aplanas** [comment]
+1

**Comment by MaximilianMeister** [comment]
just added quotes around the `"$i"`

**Comment by vuntz** [comment]
@MaximilianMeister I think we just need a fixed update-alternatives for my patch, see https://bugzilla.suse.com/show_bug.cgi?id=969171

**Comment by aplanas** [comment]
+1

File: chef/cookbooks/provisioner/templates/default/autoyast-upgrade.xml.erb [hunk 1]
```diff
@@ -108,6 +108,20 @@
...
 <% end -%>
+      <script>
+        <chrooted config:type="boolean">true</chrooted>
+        <debug config:type="boolean">true</debug>
+        <filename>fix_alternatives</filename>
+        <source>
+          <![CDATA[
+          # python-related alternatives are not updated, even though the binary
+          # names changed (2.6 to 2.7); find all broken links and fix/update them
+          for i in $(find /etc/alternatives/ -type l -! -exec test -e {} \; -print | xargs basename -a) ; do
```
Comment (nicolasbock) [inline chef/cookbooks/provisioner/templates/default/autoyast-upgrade.xml.erb]: It's generally safer to use `-print0`:

~~~
find /etc/alternatives/ -type l -! -exec test -e {} \; -print0 | xargs --null basename -a
~~~

Comment (nicolasbock) [inline chef/cookbooks/provisioner/templates/default/autoyast-upgrade.xml.erb]: I can't find documentation on `-!`. What does it do? Is it the same as `!`?

Comment (nicolasbock) [inline chef/cookbooks/provisioner/templates/default/autoyast-upgrade.xml.erb]: You can also write this without using an external command to test:
~~~
find /tmp/alternatives -type l ! -readable -printf "%f\n"
~~~

**Comment by vuntz** [comment]
Would be nice to update the `find` command as suggested, +1 after this.

**Comment by MaximilianMeister** [comment]
> Would be nice to update the find command as suggested, +1 after this.

updated.

@nicolasbock thanks for the hint on `find`

**Comment by vuntz** [comment]
+1

**Pull request merged**